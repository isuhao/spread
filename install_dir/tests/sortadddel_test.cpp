#include "common.cpp"

struct MyTest : DirInstaller
{
  MyTest(const std::string &pref = "")
    : DirInstaller(own, rules, cache, pref, true) {}

  void test(const std::string &s)
  {
    cout << "TESTING: " << s << endl;
    print(pre, "PRE");
    print(post, "POST");

    HashDir add, del;
    Hash::DirMap upgrade;
    sortAddDel(add, del, upgrade);
    print(add, "ADD");
    print(del, "DEL");
    print(upgrade, "UPGRADE");
    pre.clear();
    post.clear();
    cout << endl;
  }

  void doJob()
  {
    test("Nothing");

    post["file1"] = Hash("HASH1");
    test("Single file add");

    pre["file1"] = Hash("HASH1");
    test("Delete single file");

    post["file1"] = Hash("HASH1");
    post["file2"] = Hash("HASH2");
    pre["file3"] = Hash("HASH3");
    pre["file4"] = Hash("HASH4");
    test("Multiple adds and deletes");

    pre["file"] = Hash("OLD");
    post["file"] = Hash("NEW");
    test("Upgrade file");

    pre["file"] = Hash("OLD");
    post["file"] = Hash("OLD");
    test("Ignore file");

    pre["file1"] = Hash("OLD1");
    pre["file2"] = Hash("OLD1");
    pre["file3"] = Hash("OLD3");
    pre["file4"] = Hash("OLD4");
    pre["file5"] = Hash("OLD5");
    pre["file6"] = Hash("OLD6");
    pre["ign"] = Hash("IGN");

    post["file2"] = Hash("NEW2");
    post["file3"] = Hash("NEW2");
    post["file4"] = Hash("OLD1");
    post["file5"] = Hash("OLD5");
    post["file7"] = Hash("NEW7");
    post["ign"] = Hash("IGN");
    test("Testing everything");

    assert(!checkStatus());
    setDone();
  }
};

int main()
{
  MyTest test("some/path/");
  test.run();
  return 0;
}
