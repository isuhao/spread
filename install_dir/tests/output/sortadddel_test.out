TESTING: Nothing
PRE:
POST:
ADD:
DEL:
UPGRADE:

TESTING: Single file add
PRE:
POST:
  file1 HASH1
ADD:
  HASH1 file1
DEL:
UPGRADE:

TESTING: Delete single file
PRE:
  file1 HASH1
POST:
ADD:
DEL:
  HASH1 file1
UPGRADE:

TESTING: Multiple adds and deletes
PRE:
  file3 HASH3
  file4 HASH4
POST:
  file1 HASH1
  file2 HASH2
ADD:
  HASH1 file1
  HASH2 file2
DEL:
  HASH3 file3
  HASH4 file4
UPGRADE:

TESTING: Upgrade file
PRE:
  file OLD
POST:
  file NEW
ADD:
  NEW file
DEL:
UPGRADE:
  file OLD

TESTING: Ignore file
PRE:
  file OLD
POST:
  file OLD
ADD:
DEL:
UPGRADE:

TESTING: Testing everything
PRE:
  file1 OLD1
  file2 OLD1
  file3 OLD3
  file4 OLD4
  file5 OLD5
  file6 OLD6
  ign IGN
POST:
  file2 NEW2
  file3 NEW2
  file4 OLD1
  file5 OLD5
  file7 NEW7
  ign IGN
ADD:
  NEW2 file2
  NEW2 file3
  NEW7 file7
  OLD1 file4
DEL:
  OLD1 file1
  OLD6 file6
UPGRADE:
  file2 OLD1
  file3 OLD3
  file4 OLD4

